### PR TITLE
SLING-7321 ChildResourceViaProvider support wrapping request

### PR DESCRIPTION
**Pull request by santiagozky**

### CODE REVIEW

File: src/main/java/org/apache/sling/models/impl/via/ChildResourceViaProvider.java [hunk 1]
```diff
@@ -19,15 +19,21 @@
...
 import org.apache.sling.models.annotations.via.ChildResource;
 import org.apache.sling.models.spi.ViaProvider;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
 
 @Component
 @Service
 public class ChildResourceViaProvider implements ViaProvider {
 
+    protected final Logger log = LoggerFactory.getLogger(getClass());
```
Comment (justinedelson) [inline src/main/java/org/apache/sling/models/impl/via/ChildResourceViaProvider.java]: Please make `private static final` to match other classes.

File: src/test/java/org/apache/sling/models/impl/via/ChildResourceViaProviderTest.java [hunk 2]
```diff
@@ -0,0 +1,63 @@
+package org.apache.sling.models.impl.via;
+
+import static org.mockito.Mockito.when;
```
Comment (justinedelson) [inline src/test/java/org/apache/sling/models/impl/via/ChildResourceViaProviderTest.java]: Please move down to rest of import section

File: src/test/java/org/apache/sling/models/impl/via/ChildResourceViaProviderTest.java [hunk 3]
```diff
@@ -0,0 +1,63 @@
+package org.apache.sling.models.impl.via;
```
Comment (justinedelson) [inline src/test/java/org/apache/sling/models/impl/via/ChildResourceViaProviderTest.java]: Please move to below license to match other files.

File: src/main/java/org/apache/sling/models/impl/via/ChildResourceViaProvider.java [hunk 4]
```diff
@@ -40,8 +46,36 @@ public Object getAdaptable(Object original, String value) {
...
             return ((Resource) original).getChild(value);
+        } else if (original instanceof SlingHttpServletRequest) {
+            final SlingHttpServletRequest request = (SlingHttpServletRequest) original;
+            final Resource resource = request.getResource();
+            if (resource == null) {
+                return null;
+            }
+            Resource child = resource.getChild(value);
+            if (child == null) {
+                log.debug("could not obtain child {} of resource {}", value, resource.getPath());
```
Comment (justinedelson) [inline src/main/java/org/apache/sling/models/impl/via/ChildResourceViaProvider.java]: Please upper case "could" as "Could"

File: src/test/java/org/apache/sling/models/impl/via/ChildResourceViaProviderTest.java (hunk 3 above)
Comment (santiagozky) [reply src/test/java/org/apache/sling/models/impl/via/ChildResourceViaProviderTest.java]: done

File: src/test/java/org/apache/sling/models/impl/via/ChildResourceViaProviderTest.java (hunk 2 above)
Comment (santiagozky) [reply src/test/java/org/apache/sling/models/impl/via/ChildResourceViaProviderTest.java]: done

**Pull request merged**